src/components/Header/Header.tsx

```diff
@@ -1,21 +1,36 @@
 import style from './Header.module.scss'
+import { useState } from 'react'
 import SearchForm from './SearchForm'
+import Button from '../Button/Button'
+import Portal from '../Portal/Portal'
+import AddMovie from '../AddMovie/AddMovie'
 
-const Header = () => (
-  <div className={style.header}>
-    <div className={style.content}>
-      <div className={style.first_line}>
-        <label className={style.logo}>
-          <span>netflix</span>roulette
-        </label>
-        <a className={style.add_movie}>+ ADD MOVIE</a>
-      </div>
-      <div className={style.search_content}>
-        <h2>FIND YOUR MOVIE</h2>
-        <SearchForm />
+const Header = () => {
+
+  const [showModal, setShowModal] = useState(false)
+
+  return (
+    <div className={style.header}>
+      <div className={style.content}>
+        <div className={style.first_line}>
+          <label className={style.logo}>
+            <span>netflix</span>roulette
+          </label>
+          <Button classname={'add_movie'} title={'+ ADD MOVIE'} onClick={() => setShowModal(true)} />
+        </div>
+        <div className={style.search_content}>
+          <h2>FIND YOUR MOVIE</h2>
+          <SearchForm />
+        </div>
       </div>
+      {showModal && (
+        <Portal>
+          <AddMovie onClick={() => setShowModal(false)} />
+        </Portal>
+      )}
```

Comment:
you need to reorganize the way you render your modal. You can create your modal component that is a `Portal`. Like this you create portal in portal as I see you also render another portal within `AddMovie`